src/buffer.h

```diff
@@ -45,17 +45,37 @@ struct mpd_buffer {
 	unsigned read;
 
 	/** the actual buffer */
-	unsigned char data[4096];
+	unsigned char *data;
+
+	/** size of buffer */
+	size_t data_len;
 };
 
 /**
  * Initialize an empty buffer.
  */
-static inline void
+static inline void *
 mpd_buffer_init(struct mpd_buffer *buffer)
 {
+	const size_t std_data_len = 4096;
+
 	buffer->read = 0;
 	buffer->write = 0;
+	buffer->data_len = std_data_len;
+	buffer->data = malloc(buffer->data_len);
+	if (buffer->data != NULL)
+		memset(buffer->data, 0, buffer->data_len);
+
+	return buffer->data;
+}
+
+/**
+ * Free the buffer area.
+ */
+static inline void
+mpd_buffer_end(struct mpd_buffer *buffer)
```

Comment:
I'd prefer `_deinit()` to match with `_init()`.
Not only does this not match, it can also suggest doing a different thing, e.g. obtain a pointer to the end of the buffer/data.

Reply:
No problem. Fixed in https://github.com/MusicPlayerDaemon/libmpdclient/pull/53/commits/e76558a612f6b46325369adceb7429ae8ca8a85f